CMakeLists.txt

```diff
@@ -202,77 +202,18 @@ endif()
-            set(PY_VENV_DIR ${PROJECT_SOURCE_DIR}/.venv)
-        else()
-            set(PY_VENV_DIR ${PROJECT_SOURCE_DIR}/venv)
-        endif()
-        # Start with looking for NumPy
-        file(GLOB_RECURSE NP_COMMON_HEADER ${PY_VENV_DIR}/*npy_common.h)
-        get_filename_component(NP_HEADER_DIR ${NP_COMMON_HEADER} DIRECTORY)
-        get_filename_component(NP_INCLUDE_DIR ${NP_HEADER_DIR} DIRECTORY)
-        set(Python_NumPy_INCLUDE_DIR ${NP_INCLUDE_DIR})
-        message(INFO " Found numpy include path  ${Python_NumPy_INCLUDE_DIR}")
-        # Also try to get the real python root
-        message(INFO " Before Python exec  ${PYTHON_EXECUTABLE}")
-        get_filename_component(PYTHON_EXECUTABLE ${PY_VENV_DIR}/bin/python REALPATH)
-        message(INFO " After Python exec  ${PYTHON_EXECUTABLE}")
-        get_filename_component(PYTHON_BIN ${PYTHON_EXECUTABLE} DIRECTORY)
-        get_filename_component(PY_ROOT_DIR ${PYTHON_BIN} DIRECTORY)
-        set(Python_ROOT_DIR "${PY_ROOT_DIR}")
-        message(INFO " Python root dir to search is  ${Python_ROOT_DIR}")
-        set(Python_FIND_VIRTUALENV ONLY)
-
-        # use, i.e. don't skip the full RPATH for the build tree
-        #set(CMAKE_SKIP_BUILD_RPATH FALSE)
-        #set(CMAKE_SKIP_BUILD_RPATH TRUE)
-        #set(CMAKE_INSTALL_RPATH "/Library/Developer/CommandLineTools/Library/Frameworks")
-        #message(INFO " ORIGIN is ${ORIGIN}")
-        #SET(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
-
-        # add the automatically determined parts of the RPATH
-        # which point to directories outside the build tree to the install RPATH
-        #set(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
-        #set(CMAKE_INSTALL_RPATH_USE_LINK_PATH FALSE)
-
-        # the RPATH to be used when installing, but only if it's not a system directory
-        #list(FIND CMAKE_PLATFORM_IMPLICIT_LINK_DIRECTORIES "${CMAKE_INSTALL_PREFIX}/lib" isSystemDir)
-        #if("${isSystemDir}" STREQUAL "-1")
-        #    set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
-        #endif("${isSystemDir}" STREQUAL "-1")
-
-        # Alternatively, if there is something in the ENV for the user, try that
-    elseif(DEFINED ENV{Python_NumPy_INCLUDE_DIR})
-        set(Python_NumPy_INCLUDE_DIR $ENV{Python_NumPy_INCLUDE_DIR})
-    endif()
-
-#    if(DEFINED ENV{PYTHON_INCLUDE_DIR})
-#        set(PYTHON_INCLUDE_DIR $ENV{PYTHON_INCLUDE_DIR})
-#        message(INFO " Found 'PYTHON_INCLUDE_DIR' from ENV ($ENV{PYTHON_INCLUDE_DIR})")
-#    else()
-#        message(INFO " Did not find 'PYTHON_INCLUDE_DIR' in ENV")
-#    endif()
-#    if(DEFINED ENV{PYTHON_LIBRARIES})
-#        set(PYTHON_LIBRARIES $ENV{PYTHON_LIBRARIES})
-#        message(INFO " Found 'PYTHON_LIBRARIES' from ENV ($ENV{PYTHON_LIBRARIES})")
-#    else()
-#        message(INFO " Did not find 'PYTHON_LIBRARIES' in ENV")
-#    endif()
-#    include_directories(${PYTHON_INCLUDE_DIR})
 
     find_package(Python 3.6.8 REQUIRED COMPONENTS Interpreter Development NumPy)
```

Comment:
If the build itself isn't detecting virtual environments any more, it wouldn't hurt to test what happens if we try to regenerate a CMake build without the virtual environment still activated.  My main thought there involves IDEs that may do this for you, perhaps automatically whenever a _CMakeLists.txt_ file is edited.  We might want to add something in the docs about trouble scenarios.

Reply:
My expectation is that CMake's cache of the previous configuration will sustain it. That would need testing, which I'll do.

Reply:
Tested - configured with virtual environment active, deactivated it, `touch CMakeLists.txt` to force re-running CMake, and ran `make` and `ctest` successfully

Reply:
If some IDE is blowing away CMake configuration and regenerating it in full each time, I would see that as a problem.

Reply:
Okay, yes, I have done the above in at least one of my `launch.json` files... So, this is an "ergonomics" concern, but there's a fairly ready and sensible solution... it may break people's setups though so since it's a change, the workaround might bear mentioning in documentation somewhere.... "Note that if you are using an IDE to build ngen, you will need to activate any needed venv (or set `VIRTUAL_ENV` and `PATH` environment variables as if it was activated) as part of the build process--see your IDE's documentation for how to accomplish this." or similar.

Reply:
I'll unresolve this for now for the sake of tracking.  

Indeed, we can't account for every situation, but we can document suggestions and "gotchas" for people to pay attention to.